analysis/analysis.go

```diff
@@ -389,9 +423,13 @@ func (r *AnalysisResult) GenerateFileIndexes() []*DocstoreIndex {
 	return generateIndexEntries(r.Package, parts)
 }
 
-func (r *AnalysisResult) GenerateIPIndexes() []*DocstoreIndex {
-	// IPs are indexed as is.
-	return generateIndexEntries(r.Package, r.IPs)
+func (r *AnalysisResult) GenerateSocketIndexes() []*DocstoreIndex {
+	var parts []string
+	for _, socket := range r.Sockets {
+		parts = append(parts, fmt.Sprintf("%s-%d", socket.Address, socket.Port))
```

Comment:
I'm tempted to have this behave separately for IPv6:

e.g: [::1]:80, 127.0.0.1:80

Reply:
Seems unnecessary for now, but we can easily change this later!